src/scheduled_tasks.rs

```diff
@@ -31,111 +31,138 @@ use lemmy_utils::{
   error::{LemmyError, LemmyResult},
   REQWEST_TIMEOUT,
 };
-use reqwest::blocking::Client;
-use std::{thread, time::Duration};
+use reqwest::Client;
+use std::time::Duration;
 use tracing::{error, info, warn};
 
 /// Schedules various cleanup tasks for lemmy in a background thread
-pub fn setup(
+pub async fn setup(
```

Comment:
Now that this is async, instead of passing the DB connection, you can just pass in (or create another) DB pool.